diff --git a/node_modules/jsdom/lib/jsdom/living/css/CSSStyleDeclaration-impl.js b/node_modules/jsdom/lib/jsdom/living/css/CSSStyleDeclaration-impl.js
index d4a8eb9..738888d 100644
--- a/node_modules/jsdom/lib/jsdom/living/css/CSSStyleDeclaration-impl.js
+++ b/node_modules/jsdom/lib/jsdom/living/css/CSSStyleDeclaration-impl.js
@@ -237,7 +237,9 @@ class CSSStyleDeclarationImpl {
       }
     }
     const computedValue = this.#getComputedValue(property, value);
-    if (propertyDefinitions.has(property)) {
+    if (property.startsWith("--")) {
+      this.#cachedPropertyValues.set(property, { computedValue, value });
+    } else if (propertyDefinitions.has(property)) {
       const { longhands } = propertyDefinitions.get(property);
       if (!longhands) {
         this.#cachedPropertyValues.set(property, { computedValue, value });
